src/Http/Controllers/TicketController.php

```diff
@@ -49,36 +57,76 @@ public function store(TicketStoreRequest $request)
         $ticket->client_id = $me;
         $ticket->fill($request->validated());
         $ticket->status = $ticket->client_id == $me ? TicketStatus::UNREAD : TicketStatus::ANSWERED;
+        $changes = $ticket->changesForLog();
         $ticket->save();
-        $ticket->messages()->create([
-            'user_id' => auth()->user()->id,
+
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($ticket)
+            ->withProperties($changes)
+            ->log('created');
+
+        $message = new TicketMessage();
+        $message->fill([
+            'ticket_id' => $ticket->id,
+            'user_id' => $me,
             'message' => $request->message,
         ]);
+        $changes = $message->changesForLog();
+        $message->save();
+
+        $this->userLogger
+            ->withRequest($request)
+            ->performedOn($message)
+            ->withProperties($changes)
+            ->log('created');
+
+        // if request has file attachment, That will be store and attached to message
+        if ($request->hasfile('attachments')) {
+            foreach ($request->file('attachments') as $file) {
+                $attachment = TicketAttachment::fromUpload($file);
+                $attachment->putFile($file);
+                $attachment->save();
+            }
+        } elseif ($request->has('attachments') && is_array($request->input('attachments'))) {
+            TicketAttachment::whereIn('id', $request->input('attachments'))->update(['message_id' => $message->id]);
```

Comment:
```suggestion
            TicketAttachment::whereIn('id', $request->input('attachments'))->whereNull("message_id")->update(['message_id' => $message->id]);
```

An abusive user can use this query and change `message_id` of older attachments.